xarray/core/indexing.py

```diff
@@ -705,12 +722,7 @@ def _decompose_vectorized_indexer(indexer, shape, mode):
     Parameters
     ----------
     indexer: VectorizedIndexer
-    mode: str, one of ['basic' | 'outer' | 'outer_1vector' | 'vectorized']
-        basic: for backends that support onle basic indexer
-        outer: for backends that support basic / outer indexer
-        outer_1vector: for backends that support outer indexer inluding at most
-                       1 vector.
-        vectorized: for backends that support full vectorized indexer.
+    indexing_support: one of IndexerSupport entries
```

Comment:
IndexerSupport --> `IndexingSupport`

Having the right name will be important for searches. Also, it would be valuable if this was a reference. xarray uses sphinx, right?